omni-network/omni
# refactor(halo/voter): harden voter

Hardens the voter:
- Only start voting when local node is a validator (also vice versa)
- When starting, use latest approved height
- Restart if behind vote window (catch up if lagging behind)

TODO:
- Wire it up and start it with app context.
- Trim attestations from state that are behind vote window (they will never confirm).
- Add tests for all this.
- Rename a receiver to v.

task: none
